### PR TITLE
Couple Zalmoxis structure solver to SPIDER interior evolution

### PR DESCRIPTION
**Pull request by timlichtenberg**
## Overview

This PR couples the Zalmoxis interior structure solver to SPIDER's thermal evolution code within PROTEUS. Previously, SPIDER computed its own radial mesh using the Adams-Williamson analytical equation of state — a single-material approximation that becomes inaccurate for planets with non-Earth-like compositions or masses above about 2 Earth masses. With this PR, PROTEUS can instead use Zalmoxis to compute a physically self-consistent density-pressure-gravity profile that accounts for distinct iron core and silicate mantle layers, temperature-dependent material properties, and phase transitions. This density profile is then passed to SPIDER as an external mesh file, and SPIDER evolves the thermal state on this more accurate grid.

The coupling operates in two modes:

1. **Initial structure computation**: When a simulation starts, PROTEUS calls Zalmoxis to compute the planet's density structure, interpolates it onto SPIDER's radial grid, and writes a mesh file that SPIDER reads instead of computing its own.

2. **Structural feedback** (optional): As the planet cools and solidifies, its density changes. PROTEUS can periodically feed SPIDER's evolved temperature profile back to Zalmoxis, which recomputes the structure. The updated mesh is blended gradually into the running simulation to avoid numerical instability from sudden large changes. Structure updates are triggered by physics-based criteria: when the magma ocean temperature changes by more than 3%, when the global melt fraction changes by more than 0.05, or at a configurable ceiling interval — whichever comes first. A floor interval prevents updates from firing too frequently.

### How this connects to the other two PRs

This is one of three coordinated PRs that together enable the coupled system:

- **SPIDER PR** ([#5](https://github.com/FormingWorlds/SPIDER/pull/5)): Accepts externally-computed density profiles and adds numerical safety for extreme conditions
- **Zalmoxis PR** ([#53](https://github.com/FormingWorlds/Zalmoxis/pull/53)): Computes physically self-consistent density structures and temperature profiles for planets of arbitrary mass and composition
- **This PR (PROTEUS)**: Orchestrates the coupling between Zalmoxis and SPIDER, handling mesh interpolation, entropy remapping, gradual mesh blending, configuration synchronisation, and data management

The merge order is: **SPIDER + Zalmoxis first** (independent of each other), then **this PR** (depends on both).

### What this PR changes in practice

On `main`, PROTEUS runs SPIDER with its built-in analytical density model. With this PR, users can set `struct.module = "zalmoxis"` in their configuration file, and PROTEUS will automatically:
- Call Zalmoxis to compute the planet structure before starting the thermal evolution
- Interpolate the Zalmoxis output onto SPIDER's grid and write the mesh file
- Read the actual core-mantle boundary radius from the mesh (instead of using an analytical estimate)
- Optionally recompute the structure based on physics-based triggers as the planet cools
- Blend mesh updates gradually to prevent numerical instability
- Remap the entropy field when the mesh changes, so SPIDER can continue smoothly
- Clean up memory after each structure computation to prevent memory accumulation

## Changes

**Coupling code** (~1,900 lines across 6 files):
- `interior/spider.py`: Mesh file reading/writing/blending, entropy remapping, core-mantle boundary radius extraction from mesh, equation of state table range checking
- `interior/zalmoxis.py`: Mesh interpolation from Zalmoxis output onto SPIDER grid, adiabatic temperature mode auto-selection, prescribed temperature profile generation
- `interior/wrapper.py`: Structure update orchestration with physics-based triggers (temperature change, melt fraction change, ceiling interval), convergence tracking, memory cleanup
- `interior/common.py`: Shared data structures for interior state
- `config/_struct.py`: New configuration fields for structure update interval, mesh blending rate, convergence parameters, physics-based trigger thresholds
- `proteus.py`: Restore mesh file paths on simulation resume

**Infrastructure** (~500 lines):
- `utils/data.py`: Unified equation of state and melting curve data paths across SPIDER, Aragog, and Zalmoxis; RTPress100TPa equation of state support
- `tools/get_petsc.sh`: Fix PETSc build on RHEL 9 / Rocky Linux (CFLAGS quoting)

**Tests** (~3,300 lines across 8 test files):
- 12 SPIDER mesh tests: entropy remapping, mesh blending, convergence triggering, core-mantle boundary extraction
- 1 Zalmoxis mesh test: mesh interpolation round-trip
- 6 interior wrapper tests: structure update flow, configuration validation
- 3 integration tests: regression, structure update, smoke test
- 3 resume tests: mesh path restoration on simulation restart

**Validation scripts** (~900 lines):
- Mesh interpolation diagnostic, memory profiling wrapper, Habrok cluster job generators

## Validation

All validation tests compare against the `main` branch, where SPIDER uses its built-in Adams-Williamson analytical density model. The tests below verify that the new Zalmoxis coupling produces physically consistent results and handles edge cases robustly.

### Test 1: Mesh interpolation — is the Zalmoxis structure correctly mapped onto SPIDER's grid?

PROTEUS must interpolate Zalmoxis output (which has its own radial grid with typically hundreds of points) onto SPIDER's grid (typically 50–100 points). This test verifies that radius, pressure, density, and gravity are correctly interpolated for both cell-boundary nodes (blue circles) and cell-center nodes (red squares).

**Why this test**: If the interpolation introduces errors, SPIDER would be evolving on an incorrect density profile, defeating the purpose of the coupling.

**Result**: All four physical quantities are smoothly interpolated with no discontinuities or sign errors. Cell boundaries (circles) and cell centers (squares) are offset by half a cell width, as expected from SPIDER's staggered grid.

Mesh interpolation

### Test 2: Mesh interpolation at extreme composition — does it work for planets very different from Earth?

A planet with 3 Earth masses and only 10% iron core has a much thicker mantle than Earth (the mantle spans a much wider range of radii and pressures). This tests that the interpolation handles such extreme geometries correctly.

**Why this test**: Edge cases in interpolation often fail at extremes. A very wide mantle means the radial grid is stretched, and any errors in the interpolation would be amplified.

**Result**: Radial positions span the full mantle correctly, pressure increases monotonically to the core-mantle boundary, and gravity is negative everywhere (SPIDER convention for inward-pointing acceleration).

Extreme composition mesh

### Test 3: Radial resolution sensitivity — does the grid spacing matter?

SPIDER's default grid has 50 radial levels. To check whether this is sufficient or whether finer grids change the results, I ran the same case with 40, 50, and 150 levels.

**Why this test**: If the solution changes significantly with resolution, the default is too coarse and users need guidance on choosing an appropriate grid.

**Result**: Temperature and melt fraction evolution are nearly identical across all three resolutions, confirming that 50 levels is sufficient for this class of problems.

Resolution sensitivity

### Test 4: Analytical vs numerical density model — how do the results compare across the full parameter space?

The key comparison: does using the Zalmoxis numerical structure solver instead of SPIDER's built-in analytical density model change the thermal evolution? I ran all 9 combinations of planet mass (1, 2, 3 Earth masses) and iron core mass fraction (10%, 32.5%, 50%) with both density models — 18 simulations total on the Habrok computing cluster.

**Why this test**: This is the central scientific question. The analytical Adams-Williamson model (used on `main`) approximates the planet as a single self-compressing material. Zalmoxis (this PR) solves for the actual density structure with separate iron core and silicate mantle layers and temperature-dependent material properties. The comparison shows where and by how much the analytical approximation breaks down.

**Result**: For Earth-mass planets, both models produce similar cooling trajectories. The difference grows with planet mass: at 3 Earth masses, the Zalmoxis model produces 10–35% longer solidification times because it computes more accurate pressures and densities deep in the mantle. For extreme compositions (e.g. 3 Earth masses with only 10% iron core), the difference is most pronounced — the analytical model's single-material assumption is a poor approximation for a planet dominated by silicate mantle with a small iron core.

Analytical vs numerical — full parameter space

### Test 5: Solidification time across the parameter space

The same 18 simulations shown as a summary: solidification time (when the global melt fraction drops below 1%) vs planet mass, for both density models.

**Why this test**: Individual comparisons can be misleading. A systematic parameter study reveals whether the analytical-vs-numerical density model difference is consistent or varies with conditions.

**Result**: The Zalmoxis model consistently produces longer solidification times (10–35% depending on mass and composition). Points in the upper-right corner of the plot (around 1 million years) are 3 Earth mass planets with 10% iron core — these cases never fully solidify within the 1 million year simulation limit. At these extreme pressures (>600 GPa at the core-mantle boundary), residual deep melt persists at ~1–2% because the melting curve approaches the equation of state table boundary. This behavior is consistent across both density models and is physical rather than numerical.

Solidification time matrix

### Test 6: Mesh blending — does gradual mesh updating prevent numerical instability?

When Zalmoxis recomputes the planet structure during a running simulation, the new density profile can differ significantly from the current one. For 3 Earth mass planets, the first recomputation produces up to 63% relative shift in the radial node positions (because Zalmoxis accounts for temperature-dependent density changes that the initial structure did not). Applying this change all at once crashes the ODE solver.

The mesh blending algorithm works as follows: when a new mesh arrives from Zalmoxis, the code computes the maximum relative shift across all nodes. If this exceeds the configured limit (default: 5% of the total remaining difference), each physical quantity (radius, pressure, density, gravity) is blended linearly — the old value moves 5% toward the new value at each update step. The remaining 95% gap is carried forward and reduced by another 5% at the next step. After ~20 such convergence steps, the mesh matches the Zalmoxis solution to within machine precision. During convergence, updates fire at a shorter interval (default: 10 years instead of the normal 100-year ceiling) to resolve the blending quickly.

**Why this test**: Without blending, structural feedback crashes for all planets above 2 Earth masses. This test verifies that blending produces stable results and that the blending rate does not significantly affect the final outcome.

**Result**: All three blending rates — conservative (1% per step), default (5% per step), and effectively no blending (99% per step) — produce the same cooling trajectory within ~3%. The 1% rate simply takes more convergence steps to reach the same answer.

Mesh blending

### Test 7: Blending sensitivity and structure update frequency

I varied both the mesh blending rate and the structure re-computation interval to check sensitivity.

**Why this test**: Users need guidance on what settings to use. If results are highly sensitive to these parameters, they need to be tuned carefully; if insensitive, the defaults are robust.

**Result**: For 1 Earth mass, the blending rate has negligible effect. For 3 Earth masses, the structure update interval (50 vs 100 years) produces small but visible differences in the cooling trajectory. The defaults (5% blending, 100-year ceiling interval) are a reasonable compromise between accuracy and computational cost.

Blending sensitivity

### Test 8: Entropy remapping — does the simulation survive mesh changes?

When the radial mesh changes, SPIDER's entropy field (which is defined on the old grid) must be interpolated onto the new grid. SPIDER stores entropy internally as a gradient (dS/dξ) plus a reference value on a dimensionless coordinate. The remapping interpolates the absolute entropy profile S(r) from old to new node positions, then recomputes dS/dξ and the reference value — the exact inverse of SPIDER's forward reconstruction.

**Why this test**: The entropy field carries the full thermodynamic state of the simulation. If it is not correctly mapped to the new grid, the ODE solver crashes immediately because the old entropy gradient values make no sense on the new coordinate positions.

**Result**: The remapping allows the 3 Earth mass case (which has a 63% radius shift at the first mesh update) to survive the mesh change and continue evolving normally (56 time steps to solidification). Without remapping, this case crashed instantly after the mesh update.

### Test 9: Structural feedback — does re-computing the density structure affect thermal evolution?

This compares simulations with fixed structure (computed once at the start) against simulations where the structure is periodically recomputed as the planet cools. When the structure is recomputed, Zalmoxis receives SPIDER's current temperature profile and produces an updated density-pressure-gravity profile that reflects the changed thermal state. This captures physical effects like thermal contraction as the mantle solidifies.

**Why this test**: Structural feedback is the key new capability beyond the initial structure computation. The physical expectation is that as the mantle cools and contracts, the density increases and the convective mixing length changes, which should affect the cooling rate.

**Result**: For 1 Earth mass planets, structural feedback has minimal effect — the structure changes little during cooling at these moderate pressures. For 3 Earth masses, the feedback produces a noticeably different cooling trajectory, with faster final solidification because the contracting mantle increases convective efficiency.

Structural feedback

### Test 10: Memory usage — does garbage collection prevent memory accumulation?

Each time Zalmoxis recomputes the structure, it allocates new equation of state interpolation objects, numerical integration workspaces, and convergence history arrays. Without cleanup, this memory accumulates over hundreds of update cycles. During early development of the structural feedback feature, these simulations required 3–5 GB of memory and sometimes triggered out-of-memory kills on the Habrok cluster (which allocates 3 GB per job by default).

**Why this test**: Memory accumulation is a practical barrier to running long simulations with structural feedback, especially on shared computing clusters with memory limits.

**Result**: With explicit garbage collection (`gc.collect()`) after each structure update, all cases stay below 1.1 GB — a 3–4× reduction from the early development version.

Memory usage

### Test 11: Memory before vs after garbage collection fix

Direct comparison of peak memory usage between the early development version (without garbage collection after Zalmoxis calls, left) and the final version (with garbage collection, right). The early version accumulated SciPy interpolation objects and numerical workspaces from each Zalmoxis call, growing to 3.6–4.4 GB for runs with structure updates every 100 years. The fix adds a single `gc.collect()` call after each Zalmoxis computation, which breaks the circular references keeping these objects alive.

Memory comparison

### Test 12: Resolution convergence — are results stable across grid resolutions?

I ran 1 Earth mass simulations with 40, 90, and 120 radial levels (a 3× range) to check whether the solidification time is converged.

**Why this test**: If results change significantly with resolution, the default grid is too coarse and all results are resolution-dependent.

**Result**: Solidification times vary by less than 3% across the full resolution range, confirming convergence.

Resolution convergence

### Test 13: Radial profiles — how does the interior evolve over time?

Temperature and melt fraction profiles at the start (dashed) and end (solid) of each simulation, across five cases: 1 Earth mass with both density models, 2 and 3 Earth masses with the numerical structure solver, and 3 Earth masses with a small iron core. This shows how the mantle cools and solidifies from the base upward (the base solidifies first because it is at higher pressure, where the melting point is higher).

**Why this test**: Radial profiles provide a qualitative sanity check that the physics behaves as expected — solidification front propagating upward, temperature decreasing over time, melt fraction going to zero — and reveal structural differences between the density models and across planet masses.

**Result**: All cases show the expected bottom-up solidification pattern. More massive planets have higher interior temperatures and pressures, producing broader radial profiles. The 3 Earth mass case with 10% iron core (bottom row) has the widest mantle and shows residual deep melt even at the end of the simulation, consistent with the long solidification times seen in Test 5.

Radial profiles

### Test 14: Core-mantle boundary radius fix

During development of this coupling, I discovered that PROTEUS was passing the analytical core radius fraction (53.4%, derived from the Adams-Williamson model) to SPIDER's `-coresize` parameter even when using a Zalmoxis mesh where the actual core fraction is different (50.3% for 1 Earth mass with 32.5% iron core). This mismatch caused SPIDER's mixing-length convection model to compute zero mixing length for the bottom nodes of the mantle (those between the actual core-mantle boundary and the phantom analytical boundary), resulting in zero eddy diffusivity and zero convective heat transport in that region. The core could not cool, producing a systematic divergence between the two density models that was not caused by real physical differences.

The fix reads the actual core-mantle boundary radius directly from the mesh file (the innermost radial node position divided by the outermost) instead of using the analytical estimate. This plot shows that after the fix, the analytical and numerical density models produce consistent cooling trajectories across multiple planet masses, with remaining differences attributable to the genuine physical differences between the models.

Core-mantle boundary fix

## Challenges during development

- **Entropy remapping after mesh changes**: When Zalmoxis recomputes the structure, the radial grid changes (sometimes dramatically — up to 63% for massive planets). SPIDER stores entropy as a gradient (dS/dξ) plus a reference value on a non-dimensional coordinate grid. Correctly remapping this to a new mesh required understanding SPIDER's internal coordinate system (ξ, which is dimensionless and based on the mass coordinate) and implementing the exact inverse of SPIDER's forward reconstruction formula.
- **Core-mantle boundary mismatch**: The Zalmoxis structure solver produces a different core radius fraction than the Adams-Williamson analytical model (because it accounts for temperature-dependent density). SPIDER uses this fraction to set the inner boundary of its mixing-length convection model. When the wrong value was passed, the bottom nodes of the mantle had zero mixing length, zero eddy diffusivity, and zero heat transport — a subtle bug that did not cause a crash but produced physically incorrect results (the core could not cool).
- **Memory accumulation**: Each Zalmoxis structure computation allocates equation of state interpolation objects, numerical integration workspaces, and convergence history arrays. Python's garbage collector does not free these promptly because of circular references within SciPy objects. An explicit `gc.collect()` after each structure update was needed.
- **Large mesh shifts crashing the ODE solver**: The first Zalmoxis recomputation after SPIDER has evolved the temperature significantly can produce a mesh that differs by 39–63% from the initial one (for massive planets). Applying this all at once is too large a perturbation for the ODE solver. The mesh blending algorithm with adaptive convergence was developed to handle this — it took several iterations to find blending parameters that work across the full mass range.
- **Equation of state path unification**: SPIDER, Aragog, and Zalmoxis each had their own conventions for locating equation of state data files. Unifying these paths required tracing through three different module's data loading code and harmonising the naming conventions.

## What was not tackled / future work

- **Planets above 3 Earth masses**: The current validation covers 1–3 Earth masses. Cases at 5+ Earth masses with small iron cores hit equation of state table limits at the core-mantle boundary (pressures exceeding 1 TPa). Higher-pressure equation of state tables (RTPress100TPa) are available and supported in the code, but have not been validated in coupled simulations.
- **Structural feedback for high-mass planets**: Structural feedback works well for 1–2 Earth mass planets. For 3 Earth masses, the ODE solver becomes very stiff after mesh updates due to the equation of state approaching table edges. These cases currently require very long wall-clock times (hours per time step). Improving the equation of state table coverage or the ODE solver's handling of stiff transitions would help.
- **Multi-material mantle**: The current coupling assumes a single-composition mantle (MgSiO₃). Supporting compositional layering or fractionation during solidification would require extending both Zalmoxis and the coupling code.
- **Parallel Zalmoxis calls**: Structure computation is serial. For parameter studies with many planets, parallelising the Zalmoxis calls would improve throughput on multi-core machines.

## Checklist

- [x] I have followed the [contributing guidelines](https://proteus-framework.org/PROTEUS/CONTRIBUTING.html#how-do-i-contribute)
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings or errors
- [x] I have checked that the tests still pass on my computer
- [x] I have updated the docs, as appropriate
- [x] I have added tests for these changes, as appropriate
- [x] I have checked that all dependencies have been updated, as required

### CODE REVIEW

File: src/proteus/proteus.py [hunk 1]
```diff
@@ -378,6 +388,29 @@ def start(self, *, resume: bool = False, offline: bool = False):
...
+                ) = update_structure_from_interior(
+                    self.directories,
+                    self.config,
+                    self.hf_row,
+                    self.interior_o,
+                    self.last_struct_time,
+                    self.last_struct_Tmagma,
+                    self.last_struct_Phi,
+                )
+                gc.collect()
```
[the remark this answers is not in the document]
Comment (timlichtenberg) [reply src/proteus/proteus.py]: Fixed in b324f67a — removed the duplicate gc.collect() in proteus.py. The one inside update_structure_from_interior() is sufficient.

File: tests/integration/test_smoke_zalmoxis_spider.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import numpy as np
+import pytest
+from helpers import PROTEUS_ROOT
+
+from proteus import Proteus
+
+
+@pytest.mark.smoke
+def test_smoke_zalmoxis_spider_coupling():
+    """Test Zalmoxis structure + SPIDER interior end-to-end (few timesteps).
```
[the remark this answers is not in the document]
Comment (timlichtenberg) [reply tests/integration/test_smoke_zalmoxis_spider.py]: Fixed in b324f67a — reclassified from `@pytest.mark.smoke` to `@pytest.mark.integration`. This test requires Zalmoxis + SPIDER binaries and takes 30-60s, which exceeds the smoke test <30s target.

File: tools/get_petsc.sh [hunk 3]
```diff
@@ -265,6 +278,7 @@ cd "$workpath"
     --COPTFLAGS="-g -O3" \
     $mpi_flag \
     $blas_flag \
+    ${cflags:+"--CFLAGS=$cflags"} \
     ${ldflags:+"LDFLAGS=$ldflags"}
```
[the remark this answers is not in the document]
Comment (timlichtenberg) [reply tools/get_petsc.sh]: Fixed in b324f67a — changed `--CFLAGS=` to `CFLAGS=` to match PETSc configure conventions (consistent with the existing LDFLAGS pattern).

File: src/proteus/utils/data.py [hunk 4]
```diff
@@ -977,11 +984,25 @@ def _get_sufficient(config: Config, clean: bool = False):
...
+
+    # EOS for Zalmoxis (derived from struct.zalmoxis config, not interior.eos_dir)
+    if hasattr(config, 'struct') and getattr(config.struct, 'module', None) == 'zalmoxis':
+        # Static EOS (Seager2007) — always needed for Zalmoxis core
+        download_eos_static()
+        download_Seager_EOS()
+        # Dynamic EOS — needed if mantle uses WolfBower2018
+        mantle_eos = getattr(config.struct.zalmoxis, 'mantle_eos', '')
+        if mantle_eos.startswith('WolfBower2018'):
+            download_eos_dynamic('WolfBower2018_MgSiO3')
```
[the remark this answers is not in the document]
Comment (timlichtenberg) [reply src/proteus/utils/data.py]: Fixed in b324f67a — added a `_dynamic_eos_map` dict that maps EOS prefixes to download directories. Both WolfBower2018 and RTPress100TPa are now handled.

File: src/proteus/interior/wrapper.py [hunk 5]
```diff
@@ -248,8 +306,9 @@ def run_interior(
         # Import
         from proteus.interior.spider import ReadSPIDER, RunSPIDER
 
-        # Run SPIDER
-        RunSPIDER(dirs, config, hf_all, hf_row, interior_o)
+        # Run SPIDER (pass external mesh file if available from Zalmoxis)
+        mesh_file = dirs.get('spider_mesh')
+        RunSPIDER(dirs, config, hf_all, hf_row, interior_o, mesh_file=mesh_file)
```
[the remark this answers is not in the document]
Comment (timlichtenberg) [reply src/proteus/interior/wrapper.py]: Good catch — fixed in b324f67a. The resume path now checks for `spider_mesh.dat` in the output directory and restores `dirs['spider_mesh']`, `dirs['spider_mesh_prev']`, and the convergence state flags. Validated with new Habrok Block J resume tests (jobs 27530311).

File: input/all_options.toml [hunk 6]
```diff
@@ -167,31 +167,38 @@ version = "2.0"
...
-        mantle_mass_fraction       = 0                              # mantle mass fraction [non-dim.]
-        weight_iron_frac           = 0.325                          # iron fraction in the planet [non-dim.]
-        temperature_mode           = "linear"                       # Input temperature profile choices: "isothermal", "linear", "prescribed"
-        surface_temperature        = 3500                           # Surface temperature [K], required for temperature_mode="isothermal" or "linear"
-        center_temperature         = 6000                           # Center temperature [K], required for temperature_mode="linear"
-        temperature_profile_file   = "zalmoxis_ini_input_temp.txt"  # filename with a prescribed temperature profile, required for temperature_mode="prescribed"
-        num_levels                 = 150                            # number of Zalmoxis radius layers
+        mantle_mass_fraction       = 0                              # mantle mass fraction; 0 = auto (1 - coremassfrac) for 2-layer models
+        temperature_mode           = "adiabatic"                     # "isothermal", "linear", "prescribed", "adiabatic"
```
Comment (planetmariana) [inline input/all_options.toml]: In SPIDER the adiabatic profile is set from the adiabatic gradient, in aragog is an input, maybe that's for another issue, but they should be consistent in both sides, or at least trigger a warning of verifying that. [check here](https://github.com/FormingWorlds/aragog/blob/56a4276704336b0359d37b63156f6cf724a78602/aragog/core.py#L227)

File: src/proteus/utils/data.py [hunk 7]
```diff
@@ -1148,93 +1213,154 @@ def load_melting_curve(melt_file):
...
-        'mantle': {
-            # Silicate, modeled in Seager et al. (2007) using the fourth-order Birch-Murnaghan EOS fit to MgSiO3 perovskite and DFT calculations
-            'eos_file': Seager_eos_folder / 'eos_seager07_silicate.txt'
-        },
+        'core': {'eos_file': seager_folder / 'eos_seager07_iron.txt'},
+        'mantle': {'eos_file': seager_folder / 'eos_seager07_silicate.txt'},
     }
 
-    # Build the material_properties_iron_Tdep_silicate_planets dictionary for iron/silicate planets with temperature-dependent silicate mantle EOS from Wolf & Bower (2018)
+    # Iron/T-dep silicate (Seager 2007 core + Wolf & Bower 2018 mantle)
```
Comment (planetmariana) [inline src/proteus/utils/data.py]: The description sounds like the tables mentioned below are one file with the combination of those two. Maybe is a bit confusing from that side.

**Review by planetmariana — APPROVED** [review]
Thanks @timlichtenberg it looks good for merging, passing the core mass to SPIDER directly from Zalmoxis makes more sense now. Great work 🌋
